apps/laboratory/src/components/Wagmi/WagmiCreatePrivateKeySignerTest.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from 'react'
+import { Button, Stack } from '@chakra-ui/react'
+import { generatePrivateKey, privateKeyToAccount } from 'viem/accounts'
+import { useChakraToast } from '../Toast'
+import { LOCAL_SIGNER_KEY, setItem, getItem } from '../../utils/LocalStorage'
+
+export function WagmiCreatePrivateKeySignerTest() {
+  const toast = useChakraToast()
+  const [signerAddress, setSignerAddress] = useState<string | undefined>()
+
+  function onCreateNewPrivateKey() {
+    try {
+      const privateKey = generatePrivateKey()
+      setItem(LOCAL_SIGNER_KEY, privateKey)
+      const account = privateKeyToAccount(privateKey)
+      setSignerAddress(account.address)
+      toast({
+        title: 'Created new local signer',
+        description: account.address,
+        type: 'success'
+      })
+    } catch {
+      toast({
+        title: 'Failure',
+        description: 'Failed to create new signer',
+        type: 'error'
+      })
+    }
+  }
+
+  useEffect(() => {
+    const storedLocalSignerPrivateKey = getItem(LOCAL_SIGNER_KEY)
```

Comment:
We are calling `localStorage` which is not available on the server side and I'd expect this will throw an error in the Next.js environment. We might need to make this component client by adding `"use client"` in the first line of the file.

Reply:
We probably don't need the "use client" directive since `app/laboratory` uses the pages directory not the app dir in Next.js

This is also run within an effect, so I don't expect it to throw an error.

Reply:
Good points. Just to double check if it'd be problematic. Seems like we are okay with it ✅